File: src/main/java/edu/missouristate/mars/mips/instructions/syscalls/SyscallPrintIntUnsigned.java
package edu.missouristate.mars.mips.instructions.syscalls;

import edu.missouristate.mars.ProcessingException;
import edu.missouristate.mars.ProgramStatement;
import edu.missouristate.mars.mips.hardware.RegisterFile;
import edu.missouristate.mars.util.Binary;
import edu.missouristate.mars.util.SystemIO;

/**
 * Service to display integer stored in $a0 on the console as unsigned decimal.
 */

public class SyscallPrintIntUnsigned extends AbstractSyscall {
    /**
     * Build an instance of the Print Integer Unsigned syscall.  Default service number
     * is 36 and name is "PrintIntUnsigned".
     */
    public SyscallPrintIntUnsigned() {
        super(36, "PrintIntUnsigned");
    }

    /**
     * Performs syscall function to print on the console the integer stored in $a0.
     * The value is treated as unsigned.
     */
    public void simulate(ProgramStatement statement) throws ProcessingException {
        SystemIO.printString(
                Binary.unsignedIntToIntString(RegisterFile.getValue(4)));
    }
}

File: src/main/java/edu/missouristate/mars/mips/instructions/syscalls/SyscallPrintIntUnsigned.kt
package edu.missouristate.mars.mips.instructions.syscalls

import edu.missouristate.mars.ProcessingException
import edu.missouristate.mars.ProgramStatement
import edu.missouristate.mars.mips.hardware.RegisterFile.getValue
import edu.missouristate.mars.util.Binary.unsignedIntToIntString
import edu.missouristate.mars.util.SystemIO

/**
 * Service to display integer stored in $a0 on the console as unsigned decimal.
 */
class SyscallPrintIntUnsigned : AbstractSyscall(36, "PrintIntUnsigned") {
    /**
     * Performs syscall function to print on the console the integer stored in $a0.
     * The value is treated as unsigned.
     */
    @Throws(ProcessingException::class)
    override fun simulate(statement: ProgramStatement) {
        SystemIO.printString(unsignedIntToIntString(getValue(4)))
    }
}